src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1987,6 +1998,39 @@ BuildRoutesForInsert(Query *query, DeferredErrorMessage **planningError)
 
 
 /*
+ * ExtractDistributedInsertValuesRTE does precisely that. If the provided
+ * query is not an INSERT, or if the INSERT does not have a VALUES RTE
+ * (i.e. it is not a multi-row INSERT), this function returns NULL.
+ * If all those conditions are met, an RTE representing the multiple values
+ * of a multi-row INSERT is returned.
+ */
+RangeTblEntry *
+ExtractDistributedInsertValuesRTE(Query *query)
```

Comment:
Changed the implementation of `IsMultiRowInsert` to `return ExtractDistributedInsertValuesRTE(query) != NULL` to preserve readability and the function comment.